generator/src/main/java/com/scottlogic/deg/generator/generation/FieldValueSourceEvaluator.java

```diff
@@ -16,11 +16,101 @@
 
 package com.scottlogic.deg.generator.generation;
 
+import com.scottlogic.deg.common.profile.Types;
 import com.scottlogic.deg.generator.fieldspecs.FieldSpec;
-import com.scottlogic.deg.generator.generation.fieldvaluesources.FieldValueSource;
+import com.scottlogic.deg.generator.fieldspecs.whitelist.DistributedSet;
+import com.scottlogic.deg.generator.generation.fieldvaluesources.*;
+import com.scottlogic.deg.generator.generation.fieldvaluesources.datetime.DateTimeFieldValueSource;
+import com.scottlogic.deg.generator.generation.string.generators.RegexStringGenerator;
+import com.scottlogic.deg.generator.generation.string.generators.StringGenerator;
+import com.scottlogic.deg.generator.restrictions.*;
+import com.scottlogic.deg.generator.restrictions.linear.LinearRestrictions;
+import com.scottlogic.deg.generator.restrictions.linear.LinearRestrictionsFactory;
 
-import java.util.List;
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
-public interface FieldValueSourceEvaluator {
-    List<FieldValueSource> getFieldValueSources(FieldSpec fieldSpec);
-}
\ No newline at end of file
+import static com.scottlogic.deg.common.profile.Types.*;
+import static com.scottlogic.deg.generator.restrictions.linear.LinearRestrictionsFactory.createDateTimeRestrictions;
+import static com.scottlogic.deg.generator.restrictions.linear.LinearRestrictionsFactory.createNumericRestrictions;
+import static com.scottlogic.deg.generator.utils.Defaults.*;
+
+public class FieldValueSourceEvaluator {
+    private static final FieldValueSource NULL_ONLY_SOURCE = new NullOnlySource();
+
+    public FieldValueSource getFieldValueSources(Types type, FieldSpec fieldSpec){
+
+        Optional<FieldValueSource> source = getSource(type, fieldSpec);
+
+        if (!fieldSpec.isNullable()){
+            return source.get();
```

Comment:
Should this check `source.isPresent()` before calling `.get()`?

Reply:
that would be an invalid state, it should never get into here like that.

i'll change it to source.orElseThrow